common/djangoapps/third_party_auth/views.py

```diff
@@ -40,12 +40,15 @@ def saml_metadata_view(request):
     Get the Service Provider metadata for this edx-platform instance.
     You must send this XML to any Shibboleth Identity Provider that you wish to use.
     """
+    backend_name = request.GET.get('backend_name', 'tpa-saml')
+    if backend_name not in _PSA_SAML_BACKENDS:
+        raise Http404
     if not SAMLConfiguration.is_enabled(request.site):
         raise Http404
-    complete_url = reverse('social:complete', args=("tpa-saml", ))
+    complete_url = reverse('social:complete', args=(backend_name, ))
```

Comment:
For now, can we add a read-only field to the django admin for IDPs that shows the correct link to the metadata for each IDP? (Ideally with no param for the tpa-saml ones and with the backend_name param for sap)